scripts/update-headers.js

```diff
@@ -96,40 +86,35 @@ function check(files, skipWrite = true) {
 
   for (const file of files) {
     if (
-      !/\.(swift|h|m|js)$/i.test(file) ||
+      !/\.(swift|h|m|js|sh)$/i.test(file) ||
       ignoreExps.some(exp => exp.test(file)) ||
       !existsSync(file)
     ) { continue }
-    let text = readFileSync(file, 'utf8')
-    const hasCopyright = copyrightExp.test(text)
-    const hasBanner = text.match(bannerExp)
-    const hasInstructure = instructureExp.test(text)
-    const banner = appropriateBanner(file, text)
+    const text = readFileSync(file, 'utf8')
+    const [ , shebang = '', banner = '' ] = text.match(headerExp) || []
+    const hasCopyright = /copyright/i.test(banner)
+    const hasInstructure = /instructure/i.test(banner)
 
     if (!hasCopyright) {
-      text = `${banner}\n${text}`
+      const noBang = text.replace(/^(#!.*(?:\r?\n|\r)+)/, '')
+      const updated = `${shebang}${newBanner(file, banner)}${noBang}`
       if (!skipWrite) {
-        writeFileSync(file, text, 'utf8')
+        writeFileSync(file, updated, 'utf8')
       }
       replaced.push(file)
       continue
     }
 
-    if (!hasBanner) {
```

Comment:
if there was no banner, then the above no copyright case would have already been run. This was unreachable.